packages/fiori-elements-writer/src/index.ts

```diff
@@ -42,18 +44,36 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
 
     await addOdataService(basePath, feApp.service, fs);
 
+    const coercedUI5Version = semVer.coerce(feApp.ui5?.version)!;
     const templateOptions: TemplateOptions = {
-        changesPreview: feApp.ui5?.version
-            ? semVer.lt(semVer.coerce(feApp.ui5?.version)!, changesPreviewToVersion)
-            : false,
+        changesPreview: feApp.ui5?.version ? semVer.lt(coercedUI5Version, changesPreviewToVersion) : false,
         changesLoader: feApp.service.version === OdataVersion.v2
     };
 
     // Add new files from templates e.g.
     const rootTemplatesPath = join(__dirname, '..', 'templates');
     // Add templates common to all template types
-    // Common files
-    const ignore = [feApp.appOptions?.typescript ? '**/*.js' : '**/*.ts'];
+    const jsIgnoreGlob = ['**/*.ts'];
+
+    let ignore = jsIgnoreGlob;
+    if (feApp.appOptions?.typescript === true) {
```

Comment:
Should this block be refactored (move to a new function `generateIgnoreGlob`) to reduce complexity as highlighted by Sonar (16 > 15)?

Reply:
done in d28610cad53b024aa586e2df6e9c5c1210a39b47